classes/robot/crawler.php

```diff
@@ -549,53 +551,58 @@ public function process_queue($verbose = false) {
             $recentcourses = $this->get_recentcourses();
         }
 
-        // Iterate through the queue until we find an item that is a recent course, or the time runs out.
+        // Iterate through the queue.
         $cronstart = time();
         $cronstop = $cronstart + $config->maxcrontime;
-        $hasmore = true;
         $hastime = true;
-        while ($hasmore && $hastime) {
-            // Grab the first item from the queue.
-            $node = $DB->get_record_sql('SELECT *
+
+        $timeout = 0; // Lock this resource with a zero second timeout.
+        $locktype = 'tool_crawler_process_queue';
```

Comment:
same with this, why have a var which is only used once?

Reply:
Have removed this